src/modules/emote_menu/twitch/EmoteMenu.jsx

```diff
@@ -46,41 +52,70 @@ export default class EmoteMenuModule {
-    settings.on(`changed.${SettingIds.EMOTE_MENU}`, () => this.loadLegacyButton());
+    watcher.on('load.chat', () => this.loadButton());
+    settings.on(`changed.${SettingIds.EMOTE_MENU}`, () => this.loadButton());
   }
 
-  loadLegacyButton() {
+  loadButton() {
     if (getCurrentUser() == null) {
       return;
     }
 
-    const legacyContainer = document.querySelector(LEGACY_BTTV_EMOTE_PICKER_BUTTON_CONTAINER_SELECTOR);
-    const emoteMenuEnabled = settings.get(SettingIds.EMOTE_MENU);
+    const emoteMenuValue = settings.get(SettingIds.EMOTE_MENU);
+    if (emoteMenuValue !== EmoteMenuTypes.NONE) {
+      let container = null;
+
+      const buttonContainer = document.querySelector(BTTV_EMOTE_PICKER_BUTTON_CONTAINER_SELECTOR);
+      if (emoteMenuValue === EmoteMenuTypes.ENABLED && buttonContainer == null) {
+        container = document.createElement('div');
+        container.setAttribute('data-a-target', BTTV_EMOTE_PICKER_BUTTON_CONTAINER_DATA_A_TARGET);
+        const nativeEmotePickerButtonContainer = document.querySelector(EMOTE_PICKER_BUTTON_SELECTOR)?.parentElement;
+        if (nativeEmotePickerButtonContainer == null) {
+          return;
+        }
+        const chatInputIcons = nativeEmotePickerButtonContainer.parentElement;
+        if (chatInputIcons == null) {
+          return;
+        }
+        chatInputIcons.classList.add(styles.chatInputIcon);
+        chatInputIcons.appendChild(container);
+      }
+
+      const legacyContainer = document.querySelector(LEGACY_BTTV_EMOTE_PICKER_BUTTON_CONTAINER_SELECTOR);
+      if (emoteMenuValue === EmoteMenuTypes.LEGACY_ENABLED && legacyContainer == null) {
+        container = document.createElement('div');
+        container.setAttribute('data-a-target', LEGACY_BTTV_EMOTE_PICKER_BUTTON_CONTAINER_DATA_A_TARGET);
+        const chatSettingsButtonContainer = document.querySelector(CHAT_SETTINGS_BUTTON_CONTAINER_SELECTOR);
+        if (chatSettingsButtonContainer == null) {
+          return;
+        }
+        const rightContainer = chatSettingsButtonContainer.lastChild;
+        rightContainer.insertBefore(container, rightContainer.lastChild);
+      }
 
-    if (emoteMenuEnabled && legacyContainer == null) {
-      const container = document.querySelector(CHAT_SETTINGS_BUTTON_CONTAINER_SELECTOR);
       if (container == null) {
+        this.show(emoteMenuValue);
         return;
       }
-      const rightContainer = container.lastChild;
-      const buttonContainer = document.createElement('div');
-      buttonContainer.setAttribute('data-a-target', 'legacy-bttv-emote-picker-button-container');
-      rightContainer.insertBefore(buttonContainer, rightContainer.lastChild);
-
-      if (mountedRoot != null) {
-        mountedRoot.unmount();
-        isMounted = false;
+
+      const currentContainer = document.querySelector(BTTV_BUTTON_CONTAINER_SELECTOR);
+      if (currentContainer != null) {
+        currentContainer.remove();
+        if (mountedRoot != null) {
```

Comment:
we've been iterating on this specific function for a bit now, but it feels like we're going in circles. 

the goals of this function were to:
1. load the button into the DOM
2. unmount the prior react root + mount the new react root, if we loaded the button into the DOM

due to the complexities added by your changes, the function is trying to do too much at once, as it is now responsible for toggling the different buttons.

a cleaner approach may be to fully unmount the prior button, if needed and if there is one. and then mount the new button, if it is needed. we can split up this function to reduce the edge cases being ran into with this.

as written, there is a new bug introduced here where we are leaking root references, since if container is not found in the dom, we never cleanup the root